packages/pyolite-kernel/src/kernel.ts

```diff
@@ -85,6 +93,31 @@ export class PyoliteKernel extends BaseKernel implements IKernel {
     return this._ready.promise;
   }
 
+  /**
+   * Interrupt the kernel.
+   */
+  interrupt(): void {
+    if (this._interruptBuffer !== null) {
+      this._interruptBuffer[0] = 2; //Send SIGINT
+    }
+  }
+
+  private setupInterruptBuffer(): void {
+    if (!window.crossOriginIsolated) {
+      console.warn(
+        'JupyterLite is not running in a cross origin isolated context. The interrupt button will not function. Details: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements'
+      );
```

Comment:
I'd probably use a tick-delimited string and break it up over multiple lines) so it can pass the linter... and it (along with many other constants) should probably be made internationalizable (but I wouldn't worry about it on this PR).

One approach might be to link to our some of our own, perhaps-not-yet-written documentation for this, as we can put it in a little more context.

Again, the average user is not going to open up the console to see this, so we may want to put this someplace else as well, e.g. the in-app kernel-focused log console... and potentially open it here.

Reply:
Hm... it's still rather a deep dive to figure out how this might be effectively set. Some ideas:
- add something to the `docs` (maybe in `deploying.md`) about how one might set the appropriate headers
- update [serve.py](https://github.com/jupyterlite/jupyterlite/blob/main/py/jupyterlite/src/jupyterlite/addons/serve.py#L72) to set the appropriate headers so we can evaluate _something_ happening on e.g. binder
  - probably put it behind a flag, e.g. `--enable-shared-buffer`
  - if this _won't_ work over HTTP (even/especially on `localhost`), we can figure out some way to make it https with e.g. `trustme`

Reply:
I'll add some documentation to deployment.md, and look into adding a flag as you suggest. Logging to the log console seems like a good idea. Should the log be printed on kernel startup, or when the stop button is pressed?